### PR TITLE
Heap allocate HTTP_DATA_CHUNK if the size of the response is too large. 

### PR DESCRIPTION
**Pull request by jkotalik**
Follow up to #604 . Though we added the pause and resume thresholds to the OutputProducer, if we do a single large write in application code, it can still cause a huge write, causing stack overflow. There are a few ways we can mitigate this, but the simpilest is if there is a large write, heap allocate the chunks instead of stackalloc.

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.ReadWrite.cs [hunk 1]
```diff
@@ -12,6 +12,8 @@ namespace Microsoft.AspNetCore.Server.IISIntegration
 {
     internal partial class IISHttpContext
     {
+        private const int StackChunkLimit = 128;
```
Comment (pakrym) [inline src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.ReadWrite.cs]: Seems to small, even for 256k stack 1k should be fine.

Comment (pakrym) [reply src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.ReadWrite.cs]: I see, it's a stuct, how big is it?

Comment (Tratcher) [reply src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.ReadWrite.cs]: Units?

File: test/IISTestSite/Startup.cs [hunk 2]
```diff
@@ -628,5 +629,15 @@ private void TestInvalidWriteOperations(IApplicationBuilder app)
                 await context.Response.WriteAsync(success ? "Success" : "Failure");
             });
         }
+
+        private void LargeResponseFile(IApplicationBuilder app)
+        {
+            app.Run(async ctx =>
+            {
+                var tempFile = Path.GetTempFileName();
+                File.WriteAllText(tempFile, new string('a', 200000000));
```
Comment (Tratcher) [inline test/IISTestSite/Startup.cs]: delete the file afterwards?

File: src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.ReadWrite.cs (hunk 1 above)
Comment (jkotalik) [reply src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.ReadWrite.cs]: I think we can increase it. The struct size is only 16 bytes: https://github.com/aspnet/HttpSysServer/blob/231cbffe1a5cc0abce11e2d0685683ed74fc7ad5/shared/Microsoft.AspNetCore.HttpSys.Sources/NativeInterop/HttpApiTypes.cs#L78

Comment (pakrym) [reply src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.ReadWrite.cs]: 128 is good if it's a segment count.

File: test/IISTestSite/Startup.cs (hunk 2 above)
Comment (jkotalik) [reply test/IISTestSite/Startup.cs]: done.

**Comment by jkotalik** [comment]
Verifying that the old build fails with a S.O. if the write is too large.

**Comment by jkotalik** [comment]
@pakrym it definitely fails without this change.

**Pull request merged**